File: content/03/code/js/06-added-person-parameter/features/step_definitions/steps.js
const { Given, When, Then } = require('@cucumber/cucumber')
const Person = require('../../src/shouty')
require('../support/person_parameter')

Given('{person} is located/standing {int} metre(s) from Sean', function (lucy, distance) {

  console.log(lucy);

  // Write code here that turns the phrase above into concrete actions
  console.log(`Lucy is ${distance * 100} centimetres from Sean`);

  return 'pending'
})

When('Sean shouts "free bagels at Sean\'s"', function () {
  // Write code here that turns the phrase above into concrete actions
  return 'pending'
})

Then('Lucy hears Sean’s message', function () {
  // Write code here that turns the phrase above into concrete actions
  return 'pending'
})

File: content/bdd-with-cucumber/03/code/ruby/08-sketch-out-parameter-type/features/step_definitions/steps.rb
require 'shouty'

Given("{person} is located/standing {int} metre(s) from Sean") do |person, distance|
  person.move_to(distance)
end

When("Sean shouts {string}") do |message|
  @sean.shout(message)
  @message_from_sean = message
end

Then("Lucy hears Sean's message") do
  expect(@lucy.messages_heard).to eq [@message_from_sean]
end
